pkg/virt-api/validating-webhook/validating-webhook.go

```diff
@@ -759,6 +770,15 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 				}
 			}
 		}
+
+		// Validate that every network was assign to an interface
+		if len(networkInterfaceMap) != len(networkNameMap) {
```

Comment:
We could create a second map and to the check in the other direction if the length does not match, but good enough as a start.